cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamAdminModules.java

```diff
@@ -81,7 +116,7 @@ protected void configure() {
         bind(StreamMetaStore.class).to(MDSStreamMetaStore.class).in(Singleton.class);
         bind(StreamConsumerStateStoreFactory.class).to(HBaseStreamConsumerStateStoreFactory.class).in(Singleton.class);
         bind(StreamAdmin.class).to(FileStreamAdmin.class).in(Singleton.class);
-        bind(StreamConsumerFactory.class).to(HBaseStreamFileConsumerFactory.class).in(Singleton.class);
+        bind(StreamConsumerFactory.class).to(streamConsumerFactory).in(Singleton.class);
```

Comment:
Instead of passing in `cConf` and do conditional binding, it's better bind to a `Provider`, with a `CConfiguration` and `Injector` injected to the Provider and let the `Provider.get()` to make the decision. See https://github.com/caskdata/cdap/blob/develop/cdap-security/src/main/java/co/cask/cdap/security/guice/SecurityModule.java#L103
on how it is done.